File: problem_0014/inle_14/inle_14.cpp
#include <iostream>

int length_collatz(int number)
{
    int count = 1;
    std::cout << number;
    while (number != 1)
    {
        if(number % 2 == 0)
        {
            number = number/2;
        } 
        else 
        {
            number = 3 * number - 1;
        }
        count ++;
    } 
    return count;
}


int longest_collatz(int64_t max) 
{
    int length;
    int longest;
    int current_len;
    std::cout << max;
    for (int start = 2; start < max; start++)
    {
        current_len = length_collatz(start);

        std::cout << "this happened" << start;

        if (length < current_len)
        {
            length = current_len;
            longest = start;
        }
    }
    return longest;
}

int main(){
    std::cout << longest_collatz(1000000);
}

File: problem_0014/inle_14/inle_14.py
def length_collatz(number):
    count = 1
    while number > 1:
        if number%2 == 0:
            number = number/2
        else:
            number = 3 * number + 1
        count += 1
    return count

def longest_collatz(max):
    longest = 0
    length = 0
    for start in range(2, max):
        current_len = length_collatz(start)
        if length < current_len:
            length = current_len
            longest = start
    return longest

def optimized_collatz(max):
    bool_array = np.full(max, 0, dtype = int)



if __name__ == "__main__":
    print(longest_collatz(1000000))
